target/scripts/startup/setup-stack.sh

```diff
@@ -0,0 +1,1578 @@
+#! /bin/bash
+
+function setup
+{
+  _notify 'tasklog' 'Configuring mail server'
+  for FUNC in "${FUNCS_SETUP[@]}"
+  do
+    ${FUNC}
+  done
+}
+
+function _setup_supervisor
+{
+  if ! grep -q "loglevel = ${SUPERVISOR_LOGLEVEL}" /etc/supervisor/supervisord.conf
+  then
+    case "${SUPERVISOR_LOGLEVEL}" in
+      "critical" | "error" | "warn" | "info" | "debug" )
+        sed -i -E \
+          "s+loglevel.*+loglevel = ${SUPERVISOR_LOGLEVEL}+g" \
+          /etc/supervisor/supervisord.conf
+        ;;
+
+      * )
+        _notify 'error' \
+          "SUPERVISOR_LOGLEVEL value '${SUPERVISOR_LOGLEVEL}' unknown. Defaulting to 'warn'"
+
+        sed -i -E \
+          "s+loglevel.*+loglevel = warn+g" \
+          /etc/supervisor/supervisord.conf
+        ;;
+
+    esac
+
+    supervisorctl reload
+  fi
+}
+
+function _setup_default_vars
+{
+  _notify 'task' "Setting up default variables"
+
+  # update POSTMASTER_ADDRESS - must be done done after _check_hostname
+  POSTMASTER_ADDRESS="${POSTMASTER_ADDRESS:="postmaster@${DOMAINNAME}"}"
+
+  # update REPORT_SENDER - must be done done after _check_hostname
+  REPORT_SENDER="${REPORT_SENDER:="mailserver-report@${HOSTNAME}"}"
+  PFLOGSUMM_SENDER="${PFLOGSUMM_SENDER:=${REPORT_SENDER}}"
+
+  # set PFLOGSUMM_TRIGGER here for backwards compatibility
+  # when REPORT_RECIPIENT is on the old method should be used
+  # ! needs to be a string comparison
+  if [[ ${REPORT_RECIPIENT} == "0" ]]
+  then
+    PFLOGSUMM_TRIGGER="${PFLOGSUMM_TRIGGER:=none}"
+  else
+    PFLOGSUMM_TRIGGER="${PFLOGSUMM_TRIGGER:=logrotate}"
+  fi
+
+  # expand address to simplify the rest of the script
+  if [[ ${REPORT_RECIPIENT} == "0" ]] || [[ ${REPORT_RECIPIENT} == "0" ]]
+  then
+    REPORT_RECIPIENT="${POSTMASTER_ADDRESS}"
+    REPORT_RECIPIENT="${REPORT_RECIPIENT}"
+  fi
+
+  PFLOGSUMM_RECIPIENT="${PFLOGSUMM_RECIPIENT:=${REPORT_RECIPIENT}}"
+  LOGWATCH_RECIPIENT="${LOGWATCH_RECIPIENT:=${REPORT_RECIPIENT}}"
+
+  {
+    echo "export DEFAULT_RELAY_HOST='${DEFAULT_RELAY_HOST}'"
+    echo "export DMS_DEBUG='${DMS_DEBUG:-0}'"
```

Comment:
Should be `     echo "export DMS_DEBUG='${DMS_DEBUG}'"`

And line 119 removed as duplicate.

Reply:
Done.